src/frontend/screens/GpsModal.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import {View, ScrollView, StyleSheet} from 'react-native';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+  useIntl,
+} from 'react-intl';
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack';
+
+import {useLocation} from '../hooks/useLocation';
+import {useLastKnownLocation} from '../hooks/useLastSavedLocation';
+import {useLocationProviderStatus} from '../hooks/useLocationProviderStatus';
+import {usePersistedSettings} from '../hooks/persistedState/usePersistedSettings';
+import {GPS_MODAL_TEXT, WHITE} from '../lib/styles';
+import {CustomHeaderLeft} from '../sharedComponents/CustomHeaderLeft';
+import {DateDistance} from '../sharedComponents/DateDistance';
+import {FormattedCoords} from '../sharedComponents/FormattedData';
+import {Text} from '../sharedComponents/Text';
+
+const m = defineMessages({
+  gpsHeader: {
+    id: 'screens.GpsModal.gpsHeader',
+    defaultMessage: 'Current GPS Location',
+    description: 'Header for GPS screen',
+  },
+  lastUpdate: {
+    id: 'screens.GpsModal.lastUpdate',
+    defaultMessage: 'Last update',
+    description: 'Section title for time of last GPS update',
+  },
+  utm: {
+    id: 'screens.GpsModal.locationUTM',
+    defaultMessage: 'Coordinates UTM',
+    description: 'Section title for UTM coordinates',
+  },
+  dms: {
+    id: 'screens.GpsModal.locationDMS',
+    defaultMessage: 'Coordinates DMS',
+    description: 'Section title for DMS coordinates',
+  },
+  dd: {
+    id: 'screens.GpsModal.locationDD',
+    defaultMessage: 'Coordinates Decimal Degrees',
+    description: 'Section title for DD coordinates',
+  },
+  details: {
+    id: 'screens.GpsModal.details',
+    defaultMessage: 'Details',
+    description: 'Section title for details about current position',
+  },
+  yes: {
+    id: 'screens.GpsModal.yes',
+    defaultMessage: 'Yes',
+    description: 'if a location sensor is active yes/no',
+  },
+  no: {
+    id: 'screens.GpsModal.no',
+    defaultMessage: 'No',
+    description: 'if a location sensor is active yes/no',
+  },
+  locationSensors: {
+    id: 'screens.GpsModal.locationSensors',
+    defaultMessage: 'Sensor Status',
+    description: 'Heading for section about location sensor status',
+  },
+});
+
+const GpsModalRow = ({label, value}: {label: string; value: string}) => (
+  <View style={styles.row}>
+    <Text style={styles.rowLabel}>{label}</Text>
+    <Text style={styles.rowValue}>{value}</Text>
+  </View>
+);
+
+export const GpsModal = () => {
+  const {location} = useLocation({maxDistanceInterval: 15});
```

Comment:
not sure what the number here should be. using the one defined for the MapScreen